mathjax3-ts/input/tex/braket/BraketMappings.ts

```diff
@@ -0,0 +1,56 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2018 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Mappings for TeX parsing of the braket package.
+ *                                            
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+import {CommandMap, MacroMap} from '../SymbolMap.js';
+import BraketMethods from './BraketMethods.js';
+
+
+/**
+ * Macros for braket package.
+ */
+new CommandMap('Braket-macros', {
+  bra: ['Macro', '{\\langle {#1} \\vert}', 1],
+  ket: ['Macro', '{\\vert {#1} \\rangle}', 1],
```

Comment:
Are the outer braces necessary in these macros (and the ones below)?  They will produce an extra `<mrow>`, and I suppose from a semantic viewpoint, that is useful.

Are the inner braces needed?  I can see that they would be useful for `\bra{x\over y}` or `\bra{\bf x}`, but are those uses that are expected?

The spaces aren't needed (even if the inner braces are removed), but I don't mind if they are kept, but I'd prefer to see consistent spacing in the macros below (and I prefer `{#1}` to `{ #1}` or `{ #1 }`) as the extra spaces just cause the parser extra work in skipping over them.

Reply:
> Are the outer braces necessary in these macros (and the ones below)? They will produce an extra `<mrow>`, and I suppose from a semantic viewpoint, that is useful.

That's the reason.

> Are the inner braces needed?

Yes for `\bra` and `\ket` they are necessary, for the reasons you've mentioned and to stay true to the TeX behaviour. For `\Bra`, `\Ket` etc. they would not be strictly necessary (due to `\left` and `\right` treatment), however I would prefer the output of all commands to be similar.

 > The spaces aren't needed 

Fully agree. Also they varied and were unbalanced. Not sure where they came from. Maybe from the original package submitted to v2.